src/runner/pythonrunner.cpp

```diff
@@ -918,6 +792,118 @@ BOOST_PYTHON_MODULE(mobase)
       .def("modsSortedByProfilePriority", bpy::pure_virtual(&IOrganizer::modsSortedByProfilePriority))
       ;
 
+  // FileTreeEntry Scope:
+  auto fileTreeEntryClass = bpy::class_<FileTreeEntry, boost::noncopyable>("FileTreeEntry", bpy::no_init);
+  {
+    
+    bpy::scope scope = fileTreeEntryClass;
+    
+    bpy::enum_<FileTreeEntry::FileTypes>("FileTypes")
+    .value("FILE_OR_DIRECTORY", FileTreeEntry::FILE_OR_DIRECTORY)
+    .value("FILE", FileTreeEntry::FILE)
+    .value("DIRECTORY", FileTreeEntry::DIRECTORY)
+    .export_values()
+    ;
+
+    fileTreeEntryClass
+
```

Comment:
As we're in this class' scope already, wouldn't calling `def` without the `.` already add these methods to it?

Reply:
I have no idea. I tried looking this up but found no answer.

The usual way of declaring in a different scope is to declare the whole class at first, like so:

```cpp
{
    bpy::scope scope = bpy::class_<>()
        .def()
        .def()
        ;

    bpy::enum_<>();
}
```

But this does not work here because some `.def` uses the enum, so trying to load the python module fails (the code compile fines).

Reply:
So, I just tested, and this seems to work. Let me know if you want me to change the declarations.

Reply:
Currently, it looks neither like the other bpy class definitions nor like a C++ class defintion, and not particularly like something in between, either. It would be prettier and probably more readable if you can try and make it look less weird somehow. This might be something that ends up helping, or it might not.

Reply:
Here is the version with `bpy::def`, tell me what you prefer:

```cpp
// FileTreeEntry Scope:
auto fileTreeEntryClass = bpy::class_<FileTreeEntry, boost::noncopyable>("FileTreeEntry", bpy::no_init);
{
    
  bpy::scope scope = fileTreeEntryClass;
    
  bpy::enum_<FileTreeEntry::FileTypes>("FileTypes")
    .value("FILE_OR_DIRECTORY", FileTreeEntry::FILE_OR_DIRECTORY)
    .value("FILE", FileTreeEntry::FILE)
    .value("DIRECTORY", FileTreeEntry::DIRECTORY)
    .export_values()
  ;

  bpy::def("isFile", &FileTreeEntry::isFile);
  bpy::def("isDir", &FileTreeEntry::isDir);
  // Forcing the conversion to FileTypeS to avoid having to expose FileType in python:
  bpy::def("fileType", +[](FileTreeEntry* p) { return FileTreeEntry::FileTypes{ p->fileType() }; });
  // This should probably not be exposed in python since we provide automatic downcast:
  // bpy::def("getTree", static_cast<std::shared_ptr<IFileTree>(FileTreeEntry::*)()>(&FileTreeEntry::astree))
  bpy::def("name", &FileTreeEntry::name);
  bpy::def("suffix", &FileTreeEntry::suffix);
  bpy::def("time", &FileTreeEntry::time);
  bpy::def("parent", static_cast<std::shared_ptr<IFileTree>(FileTreeEntry::*)()>(&FileTreeEntry::parent));
  bpy::def("path", &FileTreeEntry::path, bpy::arg("sep") = "\\");
  bpy::def("pathFrom", &FileTreeEntry::pathFrom, bpy::arg("sep") = "\\");

  // Mutable operation:
  bpy::def("setTime", &FileTreeEntry::setTime);
  bpy::def("detach", &FileTreeEntry::detach);
  bpy::def("moveTo", &FileTreeEntry::moveTo);

  // Special methods:
  bpy::def("__eq__", +[](const FileTreeEntry* entry, QString other) { return entry->compare(other) == 0; });
  bpy::def("__eq__", +[](const FileTreeEntry* entry, std::shared_ptr<FileTreeEntry> other) { return entry == other.get(); });

  // Special methods for debug:
  bpy::def("__repr__", +[](const FileTreeEntry* entry) { return "FileTreeEntry(\"" + entry->name() + "\")"; });
}
```

It's also possible to declare directly:

```cpp
bpy::scope scope = bpy::class_<IFileTree, bpy::bases<FileTreeEntry>, boost::noncopyable>("IFileTree", bpy::no_init);
```

Reply:
It's definitely not *significantly* prettier...